### PR TITLE
Implemented Welsh Functionality

### PR DESCRIPTION
**Pull request by shreeniswarna**
Implemented Welsh Logic.

### CODE REVIEW

File: src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml [hunk 1]
```diff
@@ -1,4 +1,11 @@
-﻿@using Defra.PTS.Web.UI.Constants;
+﻿@using Defra.PTS.Web.UI.Constants
+@inject Microsoft.AspNetCore.Http.IHttpContextAccessor context
+
+@{
+    var locale = context.HttpContext.Request.Cookies["locale"] ?? "en";
```
Comment (rCorcoran) [inline src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml:5]: It's had to change to en-GB for GB sorry

File: src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml [hunk 2]
```diff
@@ -70,3 +77,15 @@
         </div>
     </div>
 </footer>
+
+<!-- Language toggle script -->
+<script>
+    function toggleLanguage() {
```
Comment (rCorcoran) [inline src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml:83]: Not sure why we've written this, the logic to change language is already there with the cookie? Please explain

**Review by rCorcoran — CHANGES_REQUESTED** [review]

File: src/Defra.PTS.Web.UI/Views/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -1,6 +1,14 @@
-﻿@using Defra.PTS.Web.UI.Constants;
+﻿@using Defra.PTS.Web.UI.Constants
+@inject Microsoft.AspNetCore.Http.IHttpContextAccessor context
+
+@{
+    var locale = context.HttpContext.Request.Cookies["locale"] ?? "en";
+    var toggleButtonText = (locale == "en") ? "Cymraeg" : "English";
```
Comment (rCorcoran) [inline src/Defra.PTS.Web.UI/Views/Shared/_Layout.cshtml:6]: again not sure why this is here

File: src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml (hunk 1 above)
Comment (shreeniswarna) [reply src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml]: completed

File: src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml (hunk 2 above)
Comment (shreeniswarna) [reply src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml]: this is the toggle function for Welsh Language

File: src/Defra.PTS.Web.UI/Views/Shared/_Layout.cshtml (hunk 3 above)
Comment (shreeniswarna) [reply src/Defra.PTS.Web.UI/Views/Shared/_Layout.cshtml:13]: It Retrives the locale from the cookies defaulting to 'en'.
Used ViewData to use locale elsewhere in the View if required
.

File: src/Defra.PTS.Web.UI/Views/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -1,4 +1,6 @@
-﻿@using Defra.PTS.Web.UI.Constants;
+﻿_Layout
```
Comment (rCorcoran) [inline src/Defra.PTS.Web.UI/Views/Shared/_Layout.cshtml:1]: Do we need this bit?

File: src/Defra.PTS.Web.UI/Views/Shared/_NavigationPartial.cshtml [hunk 5]
```diff
@@ -1,11 +1,28 @@
...
+        { "en-SignOut", "Sign out" },
+        { "cy-SignOut", "Allgofnodi" },
+        // { "en-TravelDocuments", "Lifelong pet travel documents" },
+        // { "cy-TravelDocuments", "Dogfennau teithio anifeiliaid anwes gydol oes" },
+        // { "en-ManageAccount", "Manage Account" },
+        // { "cy-ManageAccount", "Rheoli Cyfrif" }
+    };
+
+    var signOutText = translations[$"{locale}-SignOut"];
+    //var travelDocumentsText = translations[$"{locale}-TravelDocuments"];
```
Comment (rCorcoran) [inline src/Defra.PTS.Web.UI/Views/Shared/_NavigationPartial.cshtml:24]: If sonar stays angry at these then just delete and we can readd at a later time

**Review by rCorcoran — APPROVED** [review]

File: src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml [hunk 6]
```diff
@@ -30,21 +30,20 @@
                     @if(Thread.CurrentThread.CurrentCulture.EnglishName != "Welsh")
                     {
                         <li class="govuk-footer__inline-list-item">
-                            <a class="govuk-footer__link" href="" id="localeCy">
+                            <a class="govuk-footer__link" href="#" id="localeCy" onclick="changeLanguage('cy')">
```
Comment (rCorcoran) [inline src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml:33]: this is done in jQuery, it does not need to be done here.

File: src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml [hunk 7]
```diff
@@ -30,21 +30,20 @@
...
                                 Cymraeg
                             </a>
                         </li>
                     }
                     else
                     {
                         <li class="govuk-footer__inline-list-item">
-                            <a class="govuk-footer__link" href="" id="localeEn">
+                            <a class="govuk-footer__link" href="#" id="localeEn" onclick="changeLanguage('en-GB')">
```
Comment (rCorcoran) [inline src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml:41]: Same here

Comment (shreeniswarna) [reply src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml:41]: When clicked, it triggers the changeLanguage function with the argument 'en-GB', which stands for English language.

File: src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml (hunk 6 above)
Comment (shreeniswarna) [reply src/Defra.PTS.Web.UI/Views/Shared/_FooterPartial.cshtml:33]: When clicked, it triggers the changeLanguage function with the argument 'cy', which stands for Welsh language.

**Review by rCorcoran — CHANGES_REQUESTED** [review]

**Pull request closed without merging**